QModManager/Patcher.cs

```diff
@@ -306,25 +407,62 @@ internal static QMod LoadMod(QMod mod)
                 {
                     Console.WriteLine($"ERROR! Could not parse entry method {mod.AssemblyName} for mod {mod.DisplayName}");
                     Console.WriteLine(e.ToString());
-                    return null;
+                    return false;
                 }
                 catch (TargetInvocationException e)
                 {
                     Console.WriteLine($"ERROR! Invoking the specified entry method {mod.EntryMethod} failed for mod {mod.Id}");
                     Console.WriteLine(e.ToString());
-                    return null;
+                    return false;
                 }
                 catch (Exception e)
                 {
                     Console.WriteLine("ERROR! An unexpected error occurred!");
                     Console.WriteLine(e.ToString());
-                    return null;
+                    return false;
                 }
             }
 
             mod.Loaded = true;
 
-            return mod;
+            return true;
+        }
+
+        internal static List<QMod> GetDependenciesPresent(QMod mod)
+        {
+            if (mod == null) return null;
+
+            List<QMod> dependencies = new List<QMod>();
+
+            foreach(string dependencyId in mod.Dependencies)
+            {
+                foreach(QMod dependencyMod in foundMods)
+                {
+                    if (dependencyId == dependencyMod.Id)
+                        dependencies.Add(dependencyMod);
+                }
+            }
+
+            return dependencies;
+        }
+
+        internal static List<string> GetDependenciesMissing(QMod mod, IEnumerable<QMod> presentDependencies)
+        {
+            if (mod == null) return null;
+            if (presentDependencies == null || presentDependencies.Count() == 0) return mod.Dependencies.ToList();
+
+            List<string> dependenciesMissing = new List<string>(mod.Dependencies);
+
+            foreach(string dependencyId in mod.Dependencies)
+            {
+                foreach(QMod presentDependency in presentDependencies)
+                {
+                    if (dependencyId == presentDependency.Id)
+                        dependenciesMissing.Remove(dependencyId);
+                }
+            }
+
+            return dependenciesMissing;
         }
 
         internal static List<QMod> GetModsToLoadBefore(QMod mod)
```

Comment:
This function could be renamed to `GetLoadBefore` or something like that (same for `GetModsToLoadAfter`)